src/view-strategy.js

```diff
@@ -253,3 +254,77 @@ export class InlineViewStrategy {
     return viewEngine.loadViewFactory(entry, compileInstruction, loadContext, target);
   }
 }
+
+@viewStrategy()
+export class StaticViewStrategy {
+
+  template: string | HTMLTemplateElement;
+  dependencies: Function[] | { (): Function[] | (Function | Promise<Function[] | Record<string, Function>>)[] };
```

Comment:
I _think_ I get this based on previous discussions we had.
But it really needs some explanations and some tidy-up (notice for example `Function[]` is redundant in your return type).

If I get this right, you 100% need to explain that:
- it returns an array of promises rather than a promise for an array to easily support `[() => import("x"), () => import("y")] pattern`.
- that the `Record<string, Function>` is meant for lazy loading based on resource name, so the `Function` here is a factory unlike all other `Function` in that declaration, which are class ctor.

Reply:
We might want a factory returning a single resource, i.e. a function?
It might be a typo but it's not in the typedef above.

There are better use cases, but this: `() => [ async () => (await import("x")).MyElement ]`.

Reply:
Yes, `Record<string, any>` is for `import('x')`. That said, I just realized it's doesn't handle the case where an export is not a function. Probably we should ignore any export value that is not a function.